fury/ui/elements.py

```diff
@@ -4368,3 +4370,186 @@ def left_button_dragged(self, i_ren, _obj, _sub_component):
         self.panel.position += change
         self._click_position = click_position
         i_ren.force_render()
+
+
+class SpinBox(UI):
+    """SpinBox UI.
+    """
+
+    def __init__(self, position=(350, 400), size=(300, 100), padding=10,
+                 panel_color=(1, 1, 1), min_val=0, max_val=100,
+                 initial_val=50, step=1, textbox_width=10, textbox_height=2):
+        """Init this UI element.
+
+        Parameters
+        ----------
+        position : (int, int), optional
+            Absolute coordinates (x, y) of the lower-left corner of this
+            UI component.
+        size : (int, int), optional
+            Width and height in pixels of this UI component.
+        padding : int, optional
+            Distance between  and background.
+        panel_color : (float, float, float), optional
+            Panel color of SpinBoxUI.
+        min_val: int, optional
+            Minimum value of SpinBoxUI.
+        max_val: int, optional
+            Maximum value of SpinBoxUI.
+        initial_val: int, optional
+            Initial value of SpinBoxUI.
+        step: int, optional
+            Step value of SpinBoxUI.
+        textbox_width: int, optional
+            Width of Textbox.
+        textbox_height: int, optional
+            Height of Textbox.
+        """
+        self.panel_size = size
+        self.padding = padding
+        self.panel_color = panel_color
+        self.min_val = min_val
+        self.max_val = max_val
+        self.value = initial_val
+        self.step = step
+        self.textbox_width = textbox_width
+        self.textbox_height = textbox_height
+
+        super(SpinBox, self).__init__(position)
+
+        self.resize(size)
+
+        self.on_change = lambda ui: None
+
+    def _setup(self):
+        """Setup this UI component.
+
+        Create the SpinBoxUI with Background (Panel2D) and InputBox (TextBox2D)
+        and Increment,Decrement Button (Button2D).
+        """
+        self.panel = Panel2D(size=self.panel_size, color=self.panel_color)
+
+        self.textbox = TextBox2D(width=self.textbox_width,
+                                 height=self.textbox_height)
```

Comment:
you can use a percentage of `self.panel_size` to define width and height  here

Reply:
same as above

>The name might be ambiguous but these parameter sets how many characters should be there on a line (width) and how many lines (height) the textbox would have. Also being a required parameter for textbox maybe we can't skip.

Reply:
Same as above

> I still believe that the user should not have to define that for a spinbox. it should be automatically computed from `self.panel` like you do in `resize` function. Those values are skipped just after the `init` so the user never see this change.